shared/domains/chelonia/Secret.js

```diff
@@ -4,17 +4,17 @@ import { serdesDeserializeSymbol, serdesSerializeSymbol, serdesTagSymbol } from
 
 /* Wrapper class for secrets, which identifies them as such and prevents them
 from being logged */
-export class Secret<T> {
-  _content: T
 
+const wm = new WeakMap()
```

Comment:
What is the reason for introducing this `WeakMap` class and related changes?

Could you please explain here in a comment?

Reply:
The reason is that this approach is a more foolproof way of avoiding logging `Secret` objects, since the data won't be in the object itself. Before, I didn't use `WeakMap`, but since we're now using that for circular reference checking, I decided to also use it here.